drivers/unix/os_unix.cpp

```diff
@@ -703,10 +703,15 @@ bool OS_Unix::has_environment(const String &p_var) const {
 }
 
 String OS_Unix::get_environment(const String &p_var) const {
-	if (getenv(p_var.utf8().get_data())) {
-		return getenv(p_var.utf8().get_data());
+	const char *val = getenv(p_var.utf8().get_data());
+	if (val == nullptr) { // Not set; return empty string
+		return "";
 	}
-	return "";
+	String s;
+	if (s.parse_utf8(val) == OK) {
+		return s;
+	}
+	return String(val); // Not valid UTF-8, so return as-is
```

Comment:
```suggestion
	return String::utf8(val);
```

I'm not sure if this check is necessary, `String(val)` will work correctly only if string is Latin-1 (ASCII will be handled by UTF-8 parser), and this is an arbitrary assumption.

A proper way to check encoding is likely reading the value of `LC_CTYPE` variable. But for all modern systems it should be UTF-8.

Reply:
I copied this idea from https://github.com/godotengine/godot/blob/c8c483cf57a768110fce57e509f9b855e69d34b7/drivers/unix/os_unix.cpp#L515-L519, introduced in https://github.com/godotengine/godot/pull/62735.

If you want me to change that to the suggested, sure, but this seemed more bullet-proof.

Reply:
It's OK as is. It both cases, we only guarantee to parse UTF-8 and ASCII.